app/talk/board.cjsx

```diff
@@ -0,0 +1,183 @@
+React = require 'react'
+DiscussionPreview = require './discussion-preview'
+talkClient = require '../api/talk'
+authClient = require '../api/auth'
+CommentBox = require './comment-box'
+commentValidations = require './lib/comment-validations'
+discussionValidations = require './lib/discussion-validations'
+{getErrors} = require './lib/validations'
+Router = require 'react-router'
+ChangeListener = require '../components/change-listener'
+PromiseRenderer = require '../components/promise-renderer'
+Paginator = require './lib/paginator'
+Moderation = require './lib/moderation'
+
+module?.exports = React.createClass
+  displayName: 'TalkBoard'
+  mixins: [Router.Navigation]
+
+  getInitialState: ->
+    discussions: []
+    board: {}
+    discussionsMeta: {}
+    newDiscussionOpen: false
+    discussionValidationErrors: []
+
+  componentWillMount: ->
+    @setDiscussions()
+    @setBoard()
+
+  goToPage: (n) ->
+    @transitionTo(@props.pathname, @props.params, {page: n})
+    @setDiscussions(n)
+
+  discussionsRequest: (page) ->
+    board_id = +@props.params.board
+    talkClient.type('discussions').get({board_id, page_size: 5, page})
+
+  setDiscussions: (page = 1) ->
+    @discussionsRequest(page)
+      .then (discussions) =>
+        discussionsMeta = discussions[0]?.getMeta()
+        @setState {discussions, discussionsMeta}
+      .catch (e) =>
+        console.log "failed to get discussion", e
+
+  boardRequest: ->
+    id = +@props.params.board
+    talkClient.type('boards').get({id})
```

Comment:
Resource IDs are always strings. `Type::get`, given a string, will resolve to that resource.